docs/source/statistical-analysis.rst

```diff
@@ -0,0 +1,80 @@
+====================
+Statistical analysis
+====================
+
+As cryptographic security depens on proper use of the primitives, tests
+need to verify contents of parameters.
+
+Tlsfuzzer allows collecting some of the parameters to perform the analysis
+later.
+
+AES-GCM nonces
+==============
+
+The :term:`AES-GCM` construction in :term:`TLS` 1.2 uses explicit nonces.
+Peers select the nonce themselves and send it to their peer.
+
+Since reusing the nonce breaks the encryption, the peers must not do that.
+
+To collect the nonces sent by peer, use the
+:py:class:`~tlsfuzzer.messages.CollectNonces` node.
+Place it right after encryption negotiation: after
+:py:class:`~tlsfuzzer.expect.ExpectChangeCipherSpec` node.
+
+After executing the connection through runner, the passed in array has
+the nonces selected by the peer saved as binary strings—one for every record
+received.
+
+See the
+`test-aes-gcm-nonces.py
+<https://github.com/tomato42/tlsfuzzer/blob/master/scripts/test-aes-gcm-nonces.py>`_
+script for example how to verify that they monotonically increase.
+
+Saving cryptographic parameters
+===============================
+
+Unlike nonces, negotiation or advertising of other cryptographic parameters
+happens just once per connection.
+To save those parameters use the
+:py:class:`~tlsfuzzer.messages.CopyVariables` node.
+For full list of supported parameters see the class documentation,
+you can find definitions of the names in the :term:`TLS` :term:`RFC`\ s.
+
+As parameter this node accepts a dictionary where keys specify names of
```

Comment:
*As a parameter, this node accepts a dictionary in which, keys specify the names of

Reply:
updated, but without commas, I don't think there're right here, especially the comma after "which"